### PR TITLE
Ensure that dot stuffing is done after encoding

### LINKED ISSUE #21: quoted-printable and dot-stuffing filters applied in wrong order!
__TL;DR `kodus/mail` sometimes fails to do dot-stuffing on leading "." when sending mail via SMTP, leading to an SMTP protocol error. A possible fix is to use `stream_filter_prepend()` instead of `stream_filter_append()` in `Kodus\Mail\Writer::writeFiltered()`__

In the current implementation, quoted-printable encoding and SMTP dot-stuffing are both implemented using PHP stream filters. When calling `Kodus\Mail\SMTP\SMTPMailService::send()`, the `smtp.dot_stuffing` is appended to the output socket before `convert.quoted-printable-encode`. This is a mistake, as the SMTP dot-stuffing is a lower level concern (a part of the protocol) and must be performed as the last step.

Usually this works out fine, but sometimes `convert.quoted-printable-encode` will break up a long line, putting a single dot on a line by itself. Lines with single dots are exactly what the `smtp.dot_stuffing` filter is meant to handle, but it never sees this output.

An example of a mail body that does not work with the current `kodus/mail` is:
```php
Test mail 1234567890 1234567890 1234567890 1234567890 1234567890 1234567890.
More text.
```
After quoted printable (as configured by `kodus/mail`) it becomes:
```
Test mail 1234567890 1234567890 1234567890 1234567890 1234567890 1234567890=
.
More text.
```

I see two possible fixes:
1. Replace the use of `stream_filter_append()` by `stream_filter_prepend()` in `Kodus\Mail\Writer::writeFiltered()`. This causes the quoted-printable filter to be applied before dot-stuffing.
2. Refactor the code to not use (two) stream filters or to be more explicit about the ordering.

### PR DESCRIPTION
**Pull request by vortrixs**

### CODE REVIEW

**Review by rhl-jfm — APPROVED** [review]
Looks good!

I assume that the test fails without the code change (it is a proper regression test).

**Comment by vortrixs** [comment]
> Looks good!
> 
> I assume that the test fails without the code change (it is a proper regression test).

Yes, the test fails without the change. It does feel weird having tests without assertions but I just mirrored the other test in the file :sweat_smile:

**Comment by rhl-jfm** [comment]
You could maybe add a comment to that effect. Something like:
```php
    $service->send($message); // Throws if we fail to speak valid SMTP
```

File: tests/integration/MailServiceCest.php [hunk 1]
```diff
@@ -31,4 +33,30 @@ public function sendMail(IntegrationTester $I)
...
+        Test mail 1234567890 1234567890 1234567890 1234567890 1234567890 1234567890.
+        More text.
+        EOT;
+
+        $service = new SMTPMailService(
+            $I->createSocketConnector(), new NoAuthenticator(), "localhost"
+        );
+
+        $message = new Message(
+            new Address("blip@test.org", "Rasmus åh Schultz"),
```
Comment (thomasnordahl-dk) [inline tests/integration/MailServiceCest.php:49]: Let's not reference people who are not here in code. He has friends here still :-D

Comment (thomasnordahl-dk) [inline tests/integration/MailServiceCest.php:49]: Oh, I was about so say something about being personal, but I see this is copy/pasted from mindplay's own test examples back in the day, so this is a self-mention, so all good. :smile:

**Review by thomasnordahl-dk — APPROVED** [review]

**Pull request merged**